### PR TITLE
Add performance test framework

### PR DESCRIPTION
**Pull request by xlgao-zju**
- Add performance test framework
- Add a few test cases about PodSandbox

One thing to be discussed: How many times of the operations are suitable? 20(current)? 30? Or more?

Signed-off-by: Xianglin Gao <xlgao@zju.edu.cn>

### CODE REVIEW

**Comment by xlgao-zju** [comment]
Ping @feiskyer

File: README.md [hunk 1]
```diff
@@ -63,12 +63,14 @@ go get github.com/kubernetes-incubator/cri-tools/cmd/critctl
...
 - `--image-endpoint`, `-i`:CRI server image endpoint, default same as runtime endpoint.
 
-- `--build-dependencies`, `-b`:If we don't need to build dependencies, we can add this flag `--build-dependencies=false` or `-b=false`.
+- `--compile`, `-c`:If we don't need to build dependencies, we can add this flag `--compile=false` or `-c=false`.
 
 - `--ginkgo-flags`,`-g`:Space-separated list of arguments to pass to Ginkgo test runner.
 
 - `--focus`,`-f`:CRI e2e test will only run the test that match the focus regular expression.
 
+- `--benchmark`,`-b`:If set, critest will run benchmark.
```
Comment (feiskyer) [inline README.md]: only run benchmark?

File: cmd/critest/util.go [hunk 2]
```diff
@@ -29,23 +29,31 @@ import (
...
 	glog.Infof("Building cri-tools binaries...")
 	outputDir, err := getBuildOutputDir()
 	if err != nil {
 		glog.Fatalf("Failed to get build output directory: %v", err)
 	}
 
-	err = runCommand("go", "test", "-c", "-v", "-o", filepath.Join(outputDir, "e2e.test"), buildTargets[0])
-	if err != nil {
-		return fmt.Errorf("failed to build e2e.test %v", err)
+	if benchmark {
```
Comment (feiskyer) [inline cmd/critest/util.go]: It's better to add a variable of build targets (which is set depending on benchmark), and use same `runCommand` for it.

File: README.md (hunk 1 above)
Comment (xlgao-zju) [inline README.md]: yes, I think it's unnessery to run both benchmark and e2e.

Comment (feiskyer) [inline README.md]: OK, then add `only` to flag docs.

File: pkg/benchmark/pod.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package benchmark
+
+import (
+	"github.com/kubernetes-incubator/cri-tools/pkg/framework"
+	//"github.com/kubernetes-incubator/cri-tools/pkg/validate"
```
Comment (feiskyer) [inline pkg/benchmark/pod.go]: remove unused imports

File: README.md (hunk 1 above)
Comment (xlgao-zju) [inline README.md]: will do

File: pkg/benchmark/pod.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+			operation := b.Time("stop PodSandbox", func() {
+				err = c.StopPodSandbox(podID)
+			})
+
+			framework.ExpectNoError(err, "failed to stop PodSandbox: %v", err)
+			Expect(operation.Seconds()).Should(BeNumerically("<", 2), "stop PodSandbox shouldn't take too long.")
+
+			c.RemovePodSandbox(podID)
+		}, 20)
```
Comment (feiskyer) [inline pkg/benchmark/pod.go]: We could add a flag for this magic number

File: cmd/critest/util.go (hunk 2 above)
Comment (xlgao-zju) [inline cmd/critest/util.go]: `runCommand` is not only for building test, but also used to run other commands like building and running ginkgo. So, is it fine to add a parameter only for building test to a common function?

File: pkg/benchmark/pod.go (hunk 4 above)
Comment (xlgao-zju) [inline pkg/benchmark/pod.go]: How many times of the operations are suitable?

Comment (feiskyer) [inline pkg/benchmark/pod.go]: I think defaulting 20 is ok

File: cmd/critest/util.go (hunk 2 above)
Comment (feiskyer) [inline cmd/critest/util.go]: No, I mean 

```go

args := []string{"test", "-c", "-v", "-o", filepath.Join(outputDir, "e2e.test"), buildTargets[0]}
if benchmark {
	args = []string{"test", "-c", "-v", "-o", filepath.Join(outputDir, "benchmark.test"), buildTargets[1]}
}

err = runCommand("go", args...)
if err != nil {
	return fmt.Errorf("build test suite failed: %v", err)
}
```

**Comment by xlgao-zju** [comment]
Updated @feiskyer

**Comment by feiskyer** [comment]
Thanks. LGTM.

**Pull request merged**